### ISSUE TITLE
Logs: On entering a match all query and selecting 5-6 months, No results found message displayed on cancel query but data loads in 1-2 mins 

### ISSUE THREAD

**Issue by neha00290** [openobserve/openobserve#3947]
### Which OpenObserve functionalities are the source of the bug?

log search

### Is this a regression?

Yes

### Description

Steps
1. Enter match all query in logs
2. Select 6 months.
3. Run query and immediately click cancel query
4. Message displayed as query completed before cancel but UI displays No results found message for 1-2 mins
5. After sometime data loads on UI

**Scenario 2**
- Add match all query and switch to sql mode
- Switch on the quick mode toggle too
- Select 6 months and hit run query
- Click on cancel query immediately
- Observe the UI

### Please provide the version you discovered this bug in (check about page for version information)

```true
Version
Commit Hash
Build Date
v0.10.8-rc4
410aff4d1e4726f080b8f42230dabc8f7314b82e
2024-07-08T08:03:25Z
```

**Comment by omkarK06** [comment]
Fixed in #3705

**Comment by neha00290** [comment]
this is fixed
